nwb_conversion_tools/tools/neo/neo.py

```diff
@@ -0,0 +1,608 @@
+            f"{neo_reader._axon_info['fFileVersionNumber']}. Saving experiment as 'i_zero'..."
+        )
+    else:
+        protocol = neo_reader.read_raw_protocol()
+        n_commands = len(protocol[0])
+
+    if n_commands == 0:
+        icephys_experiment_type = "izero"
+        warnings.warn(
+            f"No command data found by neo reader in file {neo_reader.filename}. Saving experiment as 'i_zero'..."
+        )
+    else:
+        assert (
+            n_commands == n_segments
+        ), f"File contains inconsistent number of segments ({n_segments}) and commands ({n_commands})"
+
+    assert icephys_experiment_type in ["voltage_clamp", "current_clamp", "izero"], (
+        f"'icephys_experiment_type' should be 'voltage_clamp', 'current_clamp' or 'izero', but received value "
+        f"{icephys_experiment_type}"
+    )
+
+    # Check and auto-create electrodes, in case they don't existe yet on nwbfile
+    if len(nwbfile.ic_electrodes) == 0:
+        warnings.warn(
+            "When adding Icephys Recording, no Icephys Electrodes were found on nwbfile. Creating Electrodes now..."
+        )
+        add_icephys_electrode(
+            neo_reader=neo_reader,
+            nwbfile=nwbfile,
+            metadata=metadata,
+        )
+
+    if getattr(nwbfile, "intracellular_recordings", None):
+        ri = max(nwbfile.intracellular_recordings["responses"].index)
+    else:
+        ri = -1
+    if getattr(nwbfile, "icephys_simultaneous_recordings", None):
+        simultaneous_recordings_offset = len(nwbfile.icephys_simultaneous_recordings)
+    else:
+        simultaneous_recordings_offset = 0
+
+    if getattr(nwbfile, "icephys_sequential_recordings", None):
+        sessions_offset = len(nwbfile.icephys_sequential_recordings)
+    else:
+        sessions_offset = 0
+
+    relative_session_start_time = metadata["Icephys"]["Sessions"][sessions_offset]["relative_session_start_time"]
+    session_stimulus_type = metadata["Icephys"]["Sessions"][sessions_offset]["stimulus_type"]
+
+    # Sequential icephys recordings
+    simultaneous_recordings = list()
+    for si in range(n_segments):
+        # Parallel icephys recordings
+        recordings = list()
+        for ei, electrode in enumerate(
+            list(nwbfile.icephys_electrodes.values())[: len(neo_reader.header["signal_channels"]["units"])]
+        ):
+            # Starting time is the signal starting time within .abf file + time
+            # relative to first session (first .abf file)
+            ri += 1
+            starting_time = neo_reader.get_signal_t_start(block_index=0, seg_index=si)
+            starting_time = starting_time + relative_session_start_time
+
+            sampling_rate = neo_reader.get_signal_sampling_rate()
+            response_unit = neo_reader.header["signal_channels"]["units"][ei]
+            response_conversion = get_conversion_from_unit(unit=response_unit)
+            response_gain = neo_reader.header["signal_channels"]["gain"][ei]
```

Comment:
As per the previous conversation on the widgets, the conversion factor from relative neo units to Volts/Amps should be included in the NWB `gain` field; _i.e._, `nwb_gain = neo_gain * neo_unit_conversion_factor`